File: contracts/src/main/kotlin/jp/datachain/corda/ibc/ics2/ConsensusState.kt
package jp.datachain.corda.ibc.ics2

import com.google.protobuf.Any
import ibc.core.commitment.v1.Commitment
import jp.datachain.corda.ibc.types.Timestamp
import net.corda.core.serialization.CordaSerializable

@CordaSerializable
interface ConsensusState {
    val anyConsensusState: Any

    fun clientType(): ClientType // Consensus kind

    // GetRoot returns the commitment root of the consensus state,
    // which is used for key-value pair verification.
    fun getRoot(): Commitment.MerkleRoot

    // GetTimestamp returns the timestamp (in nanoseconds) of the consensus state
    fun getTimestamp(): Timestamp

    fun validateBasic()
}

File: go/x/ibc/light-clients/xx-corda/types/consensus-state.go
package types

import "github.com/cosmos/ibc-go/v4/modules/core/exported"

var _ exported.ConsensusState = (*ConsensusState)(nil)

func (*ConsensusState) ClientType() string {
	return CordaClientType
}

// which is used for key-value pair verification.
func (*ConsensusState) GetRoot() exported.Root {
	panic("not implemented")
}

func (*ConsensusState) GetTimestamp() uint64 {
	return 0
}

func (*ConsensusState) ValidateBasic() error {
	return nil
}
